### DISCUSSION TITLE
Can I test the backend load using apache jmeter?

### DISCUSSION THREAD

**Discussion by Myzik-Dev** [iTwin/itwinjs-core#2135] [Q&A] [answered]
Can I test the backend load using Apache Jmeter? 
I would like to measure the maximum objective CCU(Concurrent Users).

**Comment by usamaazhar** [comment, marked as answer]
Hi @Myzik-Dev ! Since our backends work with valid payload that is generated by a frontend app using imodeljs libraries it would be hard to use a tool like JMeter since that works with rest APIs, it would be tough to generalize the rest calls.

We do recommend an alternative though which is [k6.io](http://k6.io/).  You could record a browser session using your frontend app and than replay that session in load to measure CCU for your backend app ([K6 browser recorder](https://k6.io/docs/test-authoring/recording-a-session/browser-recorder/) , [k6 Scenarios ](https://k6.io/docs/using-k6/scenarios/))

**Reply by Myzik-Dev** [reply]
Like your answer, K6 seems to fit the purpose.
However, if it was a problem applying https, it was failed with "x509: certificate signed by unknown authority" in the part test of React. 
I will study more about the tool. 
Thank you for your answer.
